### PR TITLE
Split infobase and home into docker services

### LINKED ISSUE #1069: Separate infobase into separate docker-compose service
Subtask of #1071

### LINKED ISSUE #3750: Separate docker-compose service for misc scripts (like solr updater)
Subtask of #1071

### PR DESCRIPTION
**Pull request by cdrini**
Closes #1069
Closes #3750

Refactor: Separate out more services; web is now standalone!

NOTE: You want to look at this commit by commit.

### Technical
- Everything is now run as `openlibrary` instead of root
- Removed the `solrupdater` user; that's currently being run as `openlibrary` on the db 😬 That one needs a bit more thought.

### Testing
- [x] docker-compose build works
- [x] docker-compose up works
- [x] website loads
- [x] can edit records
- [x] edited records are picked up by home + solrupdater and added to solr
- [x] `chmod +x docker/*.sh`

### Screenshot
More services means the logs are now colour-coded! :) 

image

### Stakeholders
@cclauss @mekarpeles

### CODE REVIEW

**Comment by cclauss** [comment]
Added to commit message above:
- [ ] `chmod +x docker/*.sh`

**Comment by cclauss** [comment]
Do we need to make changes to the build recommendation?
```
docker build -t olbase:latest -f docker/Dockerfile.olbase . ; \
    docker-compose build solr ; \
    docker-compose build web
# --> 
docker build -t olbase:latest -f docker/Dockerfile.olbase . ; \
    docker-compose build home ; \
    docker-compose build infobase ; \
    docker-compose build solr ; \
    docker-compose build web
```

**Comment by cclauss** [comment]
On my machine, it seems to hang after a long series of...
```
web_1        | 0.0 (1): SELECT * FROM thing WHERE key='/type/redirect'
web_1        | 0.0 (1): SELECT * FROM thing WHERE key='/type/redirect'
web_1        | 0.0 (1): SELECT * FROM property WHERE type=17 AND name='location'
```

File: docker/ol-home-start.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  # and just use what's there
+  echo "Using local infogami"
+else
+  if [[ "$PYENV_VERSION" = 3* || -n $INFOGAMI ]]; then
+    pushd vendor/infogami
+    # Use Python 3 compatible infogami
+    git pull origin "${INFOGAMI:-master}"
+    popd
+  else
+    # Use production infogami
```
Comment (hornc) [inline docker/ol-home-start.sh:21]: Is 'production Infogami' accurate? Isn't it selecting a commit or tag rather than a running instance here?

Comment (cclauss) [reply docker/ol-home-start.sh:21]: Currently, this is the Infogami that is used on Production.  (vs. Dev, Staging, etc.)

**Comment by cdrini** [comment]
@cclauss The docs have already been update :) You just need to run `docker-compose build` now, and it build everything 🎉 

@cclauss Thanks for the fix! Gaaah, I ran `chmod u+x` instead of `+x` :/ 

@cclauss Is it still hanging for you? 😬

File: docker/ol-home-start.sh (hunk 1 above)
Comment (cdrini) [reply docker/ol-home-start.sh:21]:
```suggestion
    # Use version of infogami on production
```

**Comment by cclauss** [comment]
Still hanging -- Did a full rebuild - Same results on Py2 and Py3 with local Infogami or Infogami master...  Is it a Mac thing?  Or could I have a data problem?  I have been running all day on other branches with no similar issues.

**Review by cclauss — APPROVED** [review]
LGTM

File: docker/ol-web-start.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+python --version
+authbind --deep \
+  scripts/openlibrary-server conf/openlibrary.yml \
+  --gunicorn \
+  --reload \
+  --workers 4 \
```
Comment (mekarpeles) [inline docker/ol-web-start.sh:8]: We should consider upgrading and tailoring our gunicorn setup to use sane settings for workers + threads
https://docs.gunicorn.org/en/stable/settings.html

See: https://twitter.com/noufalibrahim/status/1297030284617986048 (gunicorn chat w/ 2 previous OL devs)

**Review by mekarpeles — APPROVED** [review]

**Pull request merged**